examples/immudb/src/test/java/ImmuDbTest.java

```diff
@@ -0,0 +1,171 @@
+import java.util.List;
+
+import com.mycompany.immudb.ImmuDbClient;
+import com.mycompany.immudb.ImmuDbClientImpl;
+import io.codenotary.immudb4j.Entry;
+import io.codenotary.immudb4j.ImmuClient;
+import io.codenotary.immudb4j.exceptions.CorruptedDataException;
+import io.codenotary.immudb4j.exceptions.VerificationException;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.utility.DockerImageName;
+
+/**
+ * Test class for the ImmuDbClient.
+ */
+public class ImmuDbTest {
+
+    // Default port for the ImmuDb server
+    private static final int IMMUDB_PORT = 3322;
+    // Default username for the ImmuDb server
+    private final String IMMUDB_USER = "immudb";
+    // Default password for the ImmuDb server
+    private final String IMMUDB_PASSWORD = "immudb";
+    // Default database name for the ImmuDb server
+    private final String IMMUDB_DATABASE = "defaultdb";
+
+    // Test container for the ImmuDb database, with the latest version of the image and exposed port
+    @ClassRule
+    public static final GenericContainer<?> immuDbContainer = new GenericContainer<>(DockerImageName.parse("codenotary/immudb:latest"))
+        .withExposedPorts(IMMUDB_PORT).withReuse(true);
+    // Interface for the ImmuDbClient
+    private ImmuDbClient immuDbClient;
+    // ImmuClient used to interact with the DB
+    private ImmuClient immuClient;
+
+    /**
+     * Set up the test, by creating the ImmuDbClient and connecting to the DB.
+     */
+    @Before
+    public void setUp() {
+        this.immuClient = ImmuClient.newBuilder()
+            .withServerUrl(immuDbContainer.getHost())
+            .withServerPort(immuDbContainer.getMappedPort(IMMUDB_PORT))
+            .build();
+        this.immuClient.login(IMMUDB_USER, IMMUDB_PASSWORD);
+        this.immuClient.useDatabase(IMMUDB_DATABASE);
+
+        // Initialize the ImmuDbClient
+        this.immuDbClient = new ImmuDbClientImpl(this.immuClient);
+    }
+
+    /**
+     * Test method for the putValue.
+     */
+    @Test
+    public void testPutValue() {
+        try {
+            this.immuDbClient.putValue("key", "value");
+        }
+        catch (CorruptedDataException e) {
+            Assert.fail();
+        }
+    }
+
+    /**
+     * Test method for the getValue.
+     */
+    @Test
+    public void testGetValue() {
+        try {
+            this.immuDbClient.putValue("test1", "test2");
+            Assert.assertEquals("test2", this.immuDbClient.getValue("test1"));
+        }
+        catch (CorruptedDataException | VerificationException e) {
+            Assert.fail();
+        }
+    }
```

Comment:
we can keep this test. Given the purpose is to demonstrate how to start the container and build the client, this test will be more than enough. Also, use `assertThat` from `Assertions` (assertj)